src/main/java/duke/tasklist/Storage.java

```diff
@@ -0,0 +1,86 @@
+package duke.tasklist;
+import java.io.*;
+import java.time.*;
+import java.util.*;
+
+import duke.Pikachu;
+import duke.tasks.*;
+
+public class Storage {
+    private Pikachu pikachu;
+    private String filePath;
+
+
+    //Constructor
+    public Storage(String filePath, Pikachu pikachu) {
+        this.filePath = filePath;
+        this.pikachu = pikachu;
+    }
+
+    //Accessor
+    public Pikachu getPikachu() {
+        return this.pikachu;
+    }
+
+    /**
+     * Reads the tasklist stored in Tasklist.txt, or creates one if it does not exist.
+     * Stores the read data into Pikachu's tasklist.
+     *
+     * @throws IOException If the content in Tasklist.txt is in the wrong format.
+     */
+    public void readTaskList() throws IOException {
+        //Create file
+        File tasklist = new File(filePath);
+        tasklist.createNewFile(); //if file already exists, will do nothing
+
+        FileReader fw = new FileReader(filePath);
+        BufferedReader br = new BufferedReader(fw);
+
+        //Reading of lines
+        String currLine = null;
+        ArrayList<Task> currInputList = this.pikachu.getInputList();
+        while ((currLine = br.readLine()) != null) {
+            //System.out.println("For debugging. Am I running too many times?");
+            String[] split = currLine.split(",");
+            if (split[0].equals("T")) {
+                ToDo t = new ToDo(split[2]);
+                currInputList.add(t);
+                if (split[1].equals("1")) t.mark();
```

Comment:
Should one-liner if-statements be used? Similar issue throughout the rest of the cases in this function.
```suggestion
                if (split[1].equals("1")) {
                    t.mark();
                }
```